core/src/main/java/feign/querymap/FieldQueryMapEncoder.java

```diff
@@ -67,11 +67,17 @@ private ObjectParamMetadata(List<Field> objectFields) {
     }
 
     private static ObjectParamMetadata parseObjectType(Class<?> type) {
-      return new ObjectParamMetadata(
-          Arrays.stream(type.getDeclaredFields())
-              .filter(field -> !field.isSynthetic())
-              .peek(field -> field.setAccessible(true))
-              .collect(Collectors.toList()));
+      List<Field> allFields = new ArrayList();
+
+      for (Class currentClass = type; currentClass != null; currentClass =
+          currentClass.getSuperclass()) {
+        Collections.addAll(allFields, currentClass.getDeclaredFields());
+      }
+
+      return new ObjectParamMetadata(allFields.stream()
+          .filter(field -> !field.isSynthetic())
+          .peek(field -> field.setAccessible(true))
```

Comment:
This is one area I'm unsure of.  This will allow serialization for all fields, regardless of access modifier.  I'm not sure we want make this decision for our users and take on the responsibility of doing so.  

My initial reaction is to avoid changing accessibility of fields and additionally filter on `isAccessible` fields.  Approaching it this way will prevent issues for users on JDK 11, as `setAccessible` is no longer considered _safe_, and we can document this behavior.  

The security implications of this are also important, but worth a separate discussion.

I do understand that this is existing logic, however; it is an important area to consider.

Thoughts?

Reply:
I agree. But changing the behavior is not affects the existing users? Also it might be written this way because of author did not want to deal serialization over getter&setters. As you said before it is another discussion.

Reply:
@velo I think we are waiting feedbacks to merge this pr. Thoughts?

Reply:
Well, I think for the `FieldQueryMap` it's ok to grab everything.

The only concern I had was the apache commons dependency, but, with that solved I'm good with it.

Reply:
Ok with me then.